irohad/ametsuchi/impl/postgres_ordering_service_persistent_state.cpp

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2018 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ametsuchi/impl/postgres_ordering_service_persistent_state.hpp"
+#include <boost/format.hpp>
+#include <boost/optional.hpp>
+#include <cstddef>
+#include "common/types.hpp"
+
+namespace iroha {
+  namespace ametsuchi {
+
+    expected::Result<std::shared_ptr<PostgresOrderingServicePersistentState>,
+                     std::string>
+    PostgresOrderingServicePersistentState::create(
+        std::string postgres_options) {
+      auto log_ =
+          logger::log("PostgresOrderingServicePersistentState:initConnection");
+      log_->info("Start storage creation");
```

Comment:
what is the purpose to combine log and result in one method? Better to log result the caller function.